File: src/main/java/no/nav/veilarboppfolging/client/digdir_krr/KrrPersonerResponseDto.java
package no.nav.veilarboppfolging.client.digdir_krr;

import lombok.Data;
import lombok.experimental.Accessors;

import java.util.Map;

@Data
@Accessors(chain = true)
public class KrrPersonerResponseDto {
    Map<String, DigdirKontaktinfo> personer;
    Map<String, String> feil;

    public KRRData assertSinglePersonToKrrData() {
        if (feil != null) throw new RuntimeException(String.format("Kunne ikke hente kontaktinfo fra KRR, feil: %s" + feil));
        if (personer == null || personer.size() != 1) throw new IllegalStateException("Fant ikke person i response fra KRR");
        var key = personer.keySet().stream().findFirst();
        if (key.isEmpty()) throw new IllegalStateException("Fant ingen keys (personidenter) i response fra KRR");
        return personer.get(key.get()).toKrrData();
    }
}

File: src/main/kotlin/no/nav/veilarboppfolging/client/digdir_client/KrrPersonerResponseDto.kt
package no.nav.veilarboppfolging.client.digdir_client

import no.nav.veilarboppfolging.client.digdir_krr.DigdirKontaktinfo
import no.nav.veilarboppfolging.client.digdir_krr.KRRData
import org.slf4j.LoggerFactory
import java.util.*

class KrrPersonerResponseDto {
    var personer: MutableMap<String?, DigdirKontaktinfo?>? = null
    var feil: MutableMap<String, String?>? = null

    companion object {
        val log = LoggerFactory.getLogger(KrrPersonerResponseDto::class.java)
    }

    fun assertSinglePersonToKrrData(): Optional<KRRData> {
        if (feil?.isNotEmpty() ?: false) {
            val personident = feil!!.keys.stream().findFirst().get()
            log.warn("Kunne ikke hente kontaktinfo fra KRR, feil: {}", feil!!.get(personident))
            return Optional.empty()
        }
        if (personer == null || personer!!.size != 1) {
            log.warn("Fant ikke person i response fra KRR")
            return Optional.empty()
        }
        val key = personer?.keys?.firstOrNull()
        return key?.let { personer?.get(it) }
            ?.let { Optional.of(it.toKrrData()) } ?: Optional.empty()
    }
}
